app/Http/Controllers/AccountsController.php

```diff
@@ -54,6 +54,8 @@ public function link() {
         } catch (\Tymon\JWTAuth\Exceptions\JWTException $e) {
             return response()->json(['token_absent'], $e->getStatusCode());
         }
+        if (strcmp($id, $user->id) != 0)
```

Comment:
This is incorrect. The id supplied refers to an account, not a user.
You should check if that account id belongs to the current user